### ISSUE TITLE
Search functionality

### ISSUE THREAD

**Issue by jadeclan** [abcd134/JJS3512#8]
To see the problems try these:
1a) type all in the search box on browse page
1b) then single click war (nothing happens)
1c) single click war again (Filter works!!!)

2a) type in all in the search box on browse page (it works)
2b) now add to the search string " q" (space then q)
2c) single click search (nothing, but lose search string - might be my debugging attempts)

3) sometimes need to click twice to clear filter (could be my computer)

**Comment by jadeclan** [comment]
We are down to 1 problem as far as I know.  To see it

1) go to browse via your preferred route
2) select a genre if you haven't already
3) then type something in the search box
4) note that the genre filter is wiped out

if you do a search first then a genre it works.

**Comment by jadeclan** [comment]
Got it.   Weird stuff here though.... on post back, it binds data then goes back to master site page????, and then gets Response.Redirected back to the Browse page !!!!  I had to put if statements in Master.site.cs to handle.   Very very odd!

So in summary (if I'm babbling, its cuz I'm tired):
I've repeated a series of if then else (identical ones) 3 times In !is postback, in the postback and on the master page.  And here's the weird part, on hitting the search button in browse with a filter on, it hits all 3 sets of if statements!!!! Yikes......
